src/lib/themes/blockHelpers.js

```diff
@@ -0,0 +1,61 @@
+import {DEFAULT_THEME, getColorsForTheme} from '.';
+
+// scratch-blocks colours has a pen property that scratch-gui uses for all extensions
+const getExtensionColors = theme => getColorsForTheme(theme).pen;
+
+/**
+ * Applies extension color theme to categories.
+ * No changes are applied if called with the default theme, allowing extensions to provide their own colors.
+ * These colors are not seen if the category provides a blockIconURI.
+ * @param {Array.<object>} dynamicBlockXML - XML for each category of extension blocks, returned from getBlocksXML
+ * in the vm runtime.
+ * @param {string} theme - Theme name
+ * @returns {Array.<object>} Dynamic block XML updated with colors.
+ */
+const injectExtensionCategoryColors = (dynamicBlockXML, theme) => {
+    // Don't do any manipulation for the default theme
+    if (theme === DEFAULT_THEME) return dynamicBlockXML;
+
+    const extensionColors = getExtensionColors(theme);
+    const parser = new DOMParser();
+    const serializer = new XMLSerializer();
```

Comment:
Is it possible to polyfill these in Node.js so this stuff can be added to the automated tests? Maybe `jest.mock` can help?

Reply:
I was on the fence about polyfilling because it was one more dependency and may not be identical to the browser. I didn't even consider mocking. That's a good idea. Added a test with `XMLSerializer.serializeToString` mocked.